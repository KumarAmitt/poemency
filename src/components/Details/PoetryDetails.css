.poetry-container {
  margin: 0.5em;
}

.poetry-card {
  border: 2px solid;
  padding: 0.5em;
  border-image-slice: 1;
  border-image-source: linear-gradient(to right top, #743ad5, #d53a9d);
}

.poetry-card-header {
  border-bottom: 1px solid #ccc;
}

.poetry-card-header h2 {
  text-align: center;
  margin-top: 0;
}

.poetry-card-header h4 {
  text-align: right;
  margin-right: 1em;
  font-style: oblique;
}


.poetry-card-body {
  margin-top: 2em;
  overflow: auto;
  text-align: center;
  font-size: 1.1em;
}

.empty-line {
  height: 1em;
}