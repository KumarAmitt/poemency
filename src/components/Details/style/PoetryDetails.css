.poetry-container {
  margin: 0.5em;
}

.poetry-card {
  border: 3px solid;
  padding: 0.5em;
  border-image-slice: 1;
  border-image-source: linear-gradient(to right top, #743ad5, #d53a9d);
}

.poetry-card-header {
  border-bottom: 1px solid #d53a9d;
}

.poetry-card-header h2 {
  text-align: center;
  margin-top: 0;
}

.poetry-card-header h4 {
  text-align: right;
  margin-right: 1em;
  font-style: oblique;
}

.homeIcon {
  color: #d53a9d;
}

.homeIcon:hover {
  color: #743ad5;
}

.poetry-card-body {
  margin-top: 2em;
  overflow: auto;
  text-align: center;
  font-size: 1.1em;
}

.empty-line {
  height: 1em;
}

.subheading {
  margin-top: 2em;
  text-align: center;
  color: #d53a9d;
}

@media (min-width: 768px) {
  .poetry-card {
    display: flex;
    justify-content: space-between;
  }

  .poetry-card-header {
    border-bottom: none;
    flex: 1;
    display: flex;
    flex-direction: column;
    justify-content: center;
    align-items: center;
    border-right: 2px solid #d53a9d;
  }

  .homeIcon {
    transform: scale(1.5);
  }

  .poetry-card-header > h2 {
    margin-top: 2em;
  }

  .poetry-card-body {
    flex: 2;
    max-height: 86vh;
    overflow: auto;
    margin-top: 0;
  }

  .poetry-card-body::-webkit-scrollbar {
    width: 5px;
    height: 5px;
  }

  .poetry-card-body::-webkit-scrollbar-thumb {
    background-color: #d53a9d;
  }

  .suggestions {
    display: flex;
    justify-content: center;
    margin: 2em 1em;
    overflow: auto;
    scrollbar-width: none;
  }

  .suggestions::-webkit-scrollbar {
    display: none;
  }

  .suggestions > .card {
    min-width: 14em;
  }
}

@media (min-width: 1080px) {
  .poetry-card-header {
    flex: 2;
    padding: 0.5em;
  }

  .poetry-card-header h2 {
    font-size: 2.2rem;
  }

  .poetry-card-body {
    flex: 4;
  }

  .suggestions {
    max-width: 1200px;
    margin: 2em auto;
  }
}
