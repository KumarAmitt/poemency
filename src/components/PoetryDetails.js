import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { getSinglePoetries, loadSinglePoetry } from '../store/poetry';

const PoetryDetails = ({ match }) => {
  const { params: { title } } = match;

  const dispatch = useDispatch();
  const poem = useSelector(getSinglePoetries);

  useEffect(() => {
    dispatch(loadSinglePoetry(title));
  }, []);

  return (
    <>
      <Link to="/" exact> Go back</Link>
      <h1>Details Page</h1>
      <h3>{title}</h3>
      <hr />
      {
        poem.map((p) => (
          <div key={p.title}>
            <h3>{p.title}</h3>
            <h6>{p.author}</h6>
            <p>{p.lines}</p>
          </div>
        ))
       }
    </>
  );
};

PoetryDetails.propTypes = {
  match: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default PoetryDetails;
