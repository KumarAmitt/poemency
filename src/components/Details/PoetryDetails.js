import React, { useEffect } from 'react';
import { Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import uniqid from 'uniqid';
import { useDispatch, useSelector } from 'react-redux';
import HomeIcon from '@material-ui/icons/Home';
import { loadUniqPoetry, getUniqPoetry, istUniqPoetryLoading } from '../../store/slicers/uniqPoetry';
import { loadPoetryByAuthor, getPoetryHub, getIsSameAuthor } from '../../store/slicers/poetryHub';
import Loading from '../utilityComponent/Loading';
import Error from '../utilityComponent/Error';
import './style/PoetryDetails.css';
import '../../stylesheets/shared/card.css';

const PoetryDetails = ({ match }) => {
  const { params: { title, author } } = match;

  const dispatch = useDispatch();
  const poem = useSelector(getUniqPoetry);
  const isPoemLoading = useSelector(istUniqPoetryLoading);
  const feat = useSelector(getPoetryHub);
  const isSameAuthor = useSelector(getIsSameAuthor(author));

  useEffect(() => {
    dispatch(loadUniqPoetry(author, title));
  }, []);

  useEffect(() => {
    dispatch(loadPoetryByAuthor(author));
  }, []);

  function renderPoetry(f) {
    window.scrollTo(0, 0);
    dispatch(loadUniqPoetry(f.author, f.title));
  }

  if (feat.status === 404) {
    return <Error />;
  }

  return (
    <>
      <div className="poetry-container">
        {
          isPoemLoading ? <Loading /> : (
            <div className="poetry-card">
              <div className="poetry-card-header">
                <Link to="/">
                  <HomeIcon className="homeIcon" />
                </Link>
                <h2>{poem.title}</h2>
                <h4>
                  - by
                  {' '}
                  {poem.author}
                </h4>
                <p>
                  Length:
                  {' '}
                  {poem.lineCount}
                  {' '}
                  Lines
                </p>
              </div>
              <div className="poetry-card-body">
                {
                  poem.lines && poem.lines.map((p) => (p === '' ? <p className="empty-line" key={uniqid()} /> : <pre key={uniqid()}>{p}</pre>))
                }
              </div>
              <div />
            </div>
          )
        }
      </div>
      <div className="subheading">
        {
          isSameAuthor ? <h2>From the same Composer</h2> : <h2>Search suggestions</h2>
        }
      </div>
      <div className="suggestions">
        {
          feat.map((f) => (
            <div key={uniqid()} className="card">
              <div className="card-body">
                <p>{f.author}</p>
                <h3>{f.title}</h3>
              </div>
              <div className="card-footer">
                <Link to={`/poetry/${f.author}/${f.title}`} className="link" onClick={() => renderPoetry(f)}>Read</Link>
                <p className="lineCount">
                  Lines:
                  {f.linecount}
                </p>
              </div>
            </div>
          ))
        }
      </div>
    </>
  );
};

PoetryDetails.propTypes = {
  match: PropTypes.objectOf(PropTypes.any).isRequired,
};

export default PoetryDetails;
