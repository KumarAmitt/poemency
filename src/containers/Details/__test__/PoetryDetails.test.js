import React from 'react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';

import configureAppStore from '../../../store/configureStore';
import PoetryDetails from '../PoetryDetails';

const store = configureAppStore();

describe('PoetryDetails', () => {
  test('should match snapshot of PoetryDetails', () => {
    const tree = renderer.create(
      <Provider store={store}>
        <BrowserRouter>
          <PoetryDetails match={{ params: { author: 'a' }, title: 't' }} />
        </BrowserRouter>
      </Provider>,
    ).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
