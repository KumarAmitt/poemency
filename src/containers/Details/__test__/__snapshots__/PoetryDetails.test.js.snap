// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PoetryDetails should match snapshot of PoetryDetails 1`] = `
Array [
  <div
    className="poetry-container"
  >
    <div
      className="poetry-card"
    >
      <div
        className="poetry-card-header"
      >
        <a
          href="/"
          onClick={[Function]}
        >
          <svg
            aria-hidden={true}
            className="MuiSvgIcon-root homeIcon"
            focusable="false"
            viewBox="0 0 24 24"
          >
            <path
              d="M10 20v-6h4v6h5v-8h3L12 3 2 12h3v8z"
            />
          </svg>
        </a>
        <h2 />
        <h4>
          - by
           
        </h4>
        <p>
          Length:
           
           
          Lines
        </p>
      </div>
      <div
        className="poetry-card-body"
      />
      <div />
    </div>
  </div>,
  <div
    className="subheading"
  >
    <h2>
      From the same Composer
    </h2>
  </div>,
  <div
    className="suggestions"
  />,
]
`;
